Assets/_Project/Scripts/UserInterface/Popups/PopupManager.cs

```diff
@@ -0,0 +1,94 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using JetBrains.Annotations;
+using PolSl.UrbanHealthPath.UserInterface.Initializers;
+using PolSl.UrbanHealthPath.UserInterface.Interfaces;
+using UnityEngine;
+
+namespace PolSl.UrbanHealthPath.UserInterface.Popups
+{
+    public class PopupManager : MonoBehaviour
+    {
+        public PopupType CurrentPopupType { get; private set; }
+
+        [Serializable] public struct Popup
+        {
+            [SerializeField] private PopupType type;
+            [SerializeField] private GameObject popupObject;
+
+            public PopupType GetPopupType()
+            {
+                return type;
+            }
+
+            public GameObject GetPopupObject()
+            {
+                return popupObject;
+            }
+        }
+
+        [SerializeField] private Popup[] popupsWithTypes;
+
+        private Dictionary<PopupType, GameObject> _popups;
+
+        private GameObject _currentPopup;
+
+        private static PopupManager _instance;
+
+        private void Awake()
+        {
+            if (_instance != null && _instance != this)
+            {
+                Destroy(this.gameObject);
+            }
+            else
+            {
+                _instance = this;
+            }
+
+            _instance._currentPopup = null;
+            _instance._popups = new Dictionary<PopupType, GameObject>();
+            foreach (var popup in popupsWithTypes)
+            {
+                _popups.Add(popup.GetPopupType(), popup.GetPopupObject());
+            }
+        }
+
+        public static PopupManager GetInstance()
+        {
+            if (_instance == null)
+            {
+                _instance = FindObjectOfType<PopupManager>();
+            }
+
+            return _instance;
+        }
+
+        public GameObject OpenPopup(PopupType popupType, Initializer initializer = null)
```

Comment:
Could we make this method generic? This way we should be able to get rid of this PopupType enum completely and this method could return the appropriate type:
`public T OpenPopup<T>(IInitializationParams initParams)`?

Reply:
No, we could not - Paulina Tried and she doesn't want to try again. Too time costly.